Bottom: 3cc09c6f5f066b971768e2a9f7536fca03c27b1b
Top:    df0f4dd322090dae539b0c604b01ba8548ec9348
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-28 04:23:02 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 48e2e4a..e92e1f7 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -308,6 +308,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
